src/main/java/org/craftercms/engine/store/s3/S3ContentStoreAdapter.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2007-2018 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.engine.store.s3;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.craftercms.core.exception.AuthenticationException;
+import org.craftercms.core.exception.InvalidContextException;
+import org.craftercms.core.exception.RootFolderNotFoundException;
+import org.craftercms.core.exception.StoreException;
+import org.craftercms.core.service.Context;
+import org.craftercms.core.store.impl.AbstractFileBasedContentStoreAdapter;
+import org.craftercms.core.store.impl.File;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.AmazonS3URI;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+
+/**
+ * Implementation of {@link org.craftercms.core.store.ContentStoreAdapter} to read files from AWS S3.
+ * @author joseross
+ */
+public class S3ContentStoreAdapter extends AbstractFileBasedContentStoreAdapter {
+
+    private static final Logger logger =
+        LoggerFactory.getLogger(org.craftercms.engine.store.s3.S3ContentStoreAdapter.class);
+
+    public static final String DELIMITER = "/";
+
+    protected AmazonS3 getClient() {
+        return AmazonS3ClientBuilder.defaultClient();
```

Comment:
The default client looks at other places not only the home dir, since the main objective is to make it container-friendly I think that is the best approach, but if you think we really need that option we can add it

Reply:
What other options are there? Remember that most sysadmins will probably have one set of server config changes in a Git repo or in another tool and when there's a need to make a change they can just propagate a couple of files to several servers. If they need to specify other config files for credentials they might not like it.